### PR TITLE
Basic R emulator

### PR DESCRIPTION
**Pull request by grdddj**
Connected with https://github.com/trezor/trezor-firmware/issues/2216

Creating a possibility to build and use `R` emulator via `TREZOR_MODEL=R make build_unix`.

It contains just some very basic layouts coming from `t1-in-core` rust layouts.
(Actually, it still uses `model_t1` rust feature, as the only thing that changed so far was `constants.rs::HEIGHT` from `64` to `128`. This is breaking the `T1` layout, but I was unable to properly create `model_tr`, there were issues with `#[cfg(feature`.)

All the `micropython` layouts defined in `core/src/trezor/ui/layouts/tr/__init__.py` are made just to work quickly, calling `trezorui2.confirm_action` and `trezorui2.confirm_text`.

It should support all the basic layouts needed to sign a `BTC` transaction via `Suite` and have basic `trezorctl` interaction.

Built on top of https://github.com/trezor/trezor-firmware/pull/2222 to be able to define and use a third `TREZOR_MODEL` - `R`.

Regarding the commit above, I have found two files where we are still using `TREZOR_MODEL == T` instead of `defined TREZOR_MODEL_T` - `core/embed/firmware/header.S ` and `core/embed/firmware/startup.S` - not sure if these files are working differently.

### CODE REVIEW

File: core/embed/unix/jpg_to_h.py
Comment (prusnak) [inline core/embed/unix/jpg_to_h.py:18]: Other files (1, T) use 12 columns, not 11.

File: core/SConscript.boardloader
Comment (prusnak) [inline core/SConscript.boardloader:100]: SConscript file is Python so we can use the following:

``` python
if TREZOR_MODEL in ['T', 'R']:
```

File: core/src/apps/base.py
Comment (prusnak) [inline core/src/apps/base.py:82]: This is not very nice. How about removing `PassPhraseEntry` from two lists above and adding the following?

``` python
if utils.MODEL == "T":
    f.capabilities.append(Capability.PassphraseEntry);
```

File: core/src/trezor/ui/__init__.py
Comment (prusnak) [inline core/src/trezor/ui/__init__.py:52]: This is Python, so we can use the following:

``` python
if utils.EMULATOR or utils.MODEL in ["1", "R"]:
```

File: core/src/apps/homescreen/homescreen.py
Comment (prusnak) [inline core/src/apps/homescreen/homescreen.py:56]: This is Python, so we can use the following:

``` python
if utils.MODEL in ["1", "R"]:
```

**Review by prusnak — CHANGES_REQUESTED** [review]
Left couple of nits

**Comment by grdddj** [comment]
> Most of the comments revolve around `TREZOR_MODEL` in `SConscript` files or `utils.MODEL` in Python sources.

Yeah, it is beneficial for the future. All the suggestions done in [133289a](https://github.com/trezor/trezor-firmware/pull/2230/commits/133289a0d10b3b5d1cf90ad48498b7849ac69faf)

File: core/src/apps/homescreen/homescreen.py
Comment (matejcik) [reply core/src/apps/homescreen/homescreen.py]: s/`[]`/`()`, tuples are preferred for static lists of options
or `in "1R"` works for single-letter models too :)

**Comment by prusnak** [comment]
Since https://github.com/trezor/trezor-firmware/pull/2222 has been (squashed and) merged, can you please rebase on top of current master?

**Comment by grdddj** [comment]
> Since #2222 has been (squashed and) merged, can you please rebase on top of current master?

Rebased and force-pushed now. It already includes the suggestion to compare tuples instead of lists.

File: core/embed/firmware/startup.S
Comment (grdddj) [reply core/embed/firmware/startup.S]: Changes in the force-push [7569eb0](https://github.com/trezor/trezor-firmware/pull/2230/commits/7569eb069e0f62fedd229eeef6b5cbf7fca773ce)

File: core/SConscript.unix
Comment (prusnak) [inline core/SConscript.unix:563]: typo?
```suggestion
            SOURCE_PY_DIR + 'trezor/enums/Monero*.py',
```

File: core/embed/rust/src/ui/model_t1/constant.rs
Comment (grdddj) [reply core/embed/rust/src/ui/model_t1/constant.rs]: Done in [01363d6](https://github.com/trezor/trezor-firmware/pull/2230/commits/01363d64d9b4e8e28fe20980c480cd4b5f1a4887) - created new `model_tr` feature. Having this feature adds some complexity into conditional compilation - sometimes it (VSCode) thinks something is defined multiple times, for example in 
```rust
// core/embed/rust/src/ui/layout/obj.rs
#[cfg(feature = "model_t1")]
use crate::ui::model_t1::constant;
#[cfg(feature = "model_tr")]
use crate::ui::model_tr::constant;
#[cfg(feature = "model_tt")]
use crate::ui::model_tt::constant;
```

, it does not know only one of those can be true during compile-time

**Review by prusnak — CHANGES_REQUESTED** [review]

File: core/SConscript.firmware
Comment (grdddj) [reply core/SConscript.firmware:564]: Thanks for noticing, it probably has a connection with the extra coma below. Both fixed in [a0820c1](https://github.com/trezor/trezor-firmware/pull/2230/commits/a0820c162b94f7422beb39273b18dd915e98d0ed)

**Comment by grdddj** [comment]
Last fixup - [8bfc3b0](https://github.com/trezor/trezor-firmware/pull/2230/commits/8bfc3b0fe73b31ca9668308bd600cf2ccc07b5f4) - fixes CI's `gen` and `style` tests.

It also renames all components in `core/embed/rust/src/ui/model_tr` folder to use `TR` instead of `T1` as copy-pasted there

File: core/embed/firmware/memory_R.ld
Comment (mmilata) [inline core/embed/firmware/memory_R.ld:1]: Will this file ever be different from memory_T.ld? Should we make it a symlink until then?

**Review by mmilata — APPROVED** [review]
Changes look good to me. A few comments inline. Did not test on hw.

File: core/embed/firmware/header.S
Comment (matejcik) [inline core/embed/firmware/header.S:29]: let's flip this, because the comments aren't strictly correct
```
#if !defined TREZOR_MODEL_1
// trezor-core header style
...
#else
// model 1 compatibility header
```

Comment (matejcik) [reply core/embed/firmware/header.S:29]: please do that in a separate commit because it is not emulator related

File: core/embed/rust/src/ui/component/base.rs
Comment (matejcik) [inline core/embed/rust/src/ui/component/base.rs:7]: instead: `#[cfg(any(feature = "model_t1", feature = "model_tr"))]`
dtto in all ButtonEvent cases

File: core/embed/firmware/memory_R.ld
Comment (prusnak) [reply core/embed/firmware/memory_R.ld:1]: +1 for symlink

File: python/src/trezorlib/models.py
Comment (grdddj) [reply python/src/trezorlib/models.py:38]: Sure. The version of `R` will be always the same as the version of `T`?

File: core/embed/rust/src/ui/component/base.rs
Comment (grdddj) [reply core/embed/rust/src/ui/component/base.rs:7]: Yeah, it would be the cleanest, but it is not possible, as the directory where the `event.rs` file lies is also different - `crate::ui::model_t1` vs `crate::ui::model_tr`.

During compilation, it would complain that in the case of `model_tr` feature the `model_t1` directory does not exist

Comment (grdddj) [reply core/embed/rust/src/ui/component/base.rs:7]: That brings the idea of putting it into some shared directory...

File: python/src/trezorlib/models.py
Comment (mmilata) [reply python/src/trezorlib/models.py:38]: IIRC there's some ongoing discussion regarding versioning but it's probably out of scope for this PR.

File: core/embed/rust/src/ui/display.rs
Comment (matejcik) [reply core/embed/rust/src/ui/display.rs:8]: @grdddj does vscode actually break on this? For me it highlighted these lines as incorrect but otherwise the analyzer features worked fine. I'm OK with that result. I would go for explicit mutual exclusion only in case this breaks in a more serious way (e.g. impossible to follow definitions)

File: core/embed/rust/src/ui/component/base.rs
Comment (matejcik) [reply core/embed/rust/src/ui/component/base.rs:7]: right, I didn't notice that, sorry.

Yes, shared definitions are the way to go. In fact, it should be something like `ui/event.rs` defining both Button and Touch events. We might not even need these to be conditional on model, as long as the usages are conditional. What do you think @mmilata ?

Comment (mmilata) [reply core/embed/rust/src/ui/component/base.rs:7]: +1 for sharing

File: core/embed/rust/src/ui/display.rs
Comment (grdddj) [reply core/embed/rust/src/ui/display.rs:8]: It does not break anything, it just complains - no real error (not even warning) during compilation (which is completely outside of VSCode).

File: python/src/trezorlib/models.py
Comment (grdddj) [reply python/src/trezorlib/models.py]: Comment removed in [5cd212d](https://github.com/trezor/trezor-firmware/pull/2230/commits/5cd212d6b092437e82f102909ec1edd9837f82a3)

File: core/embed/firmware/memory_R.ld
Comment (grdddj) [reply core/embed/firmware/memory_R.ld]: Symlink added in [5cd212d](https://github.com/trezor/trezor-firmware/pull/2230/commits/5cd212d6b092437e82f102909ec1edd9837f82a3)

File: core/embed/rust/src/ui/component/base.rs
Comment (grdddj) [reply core/embed/rust/src/ui/component/base.rs]: Shared `event.rs` created in [c042b45](https://github.com/trezor/trezor-firmware/pull/2230/commits/c042b4567c0ced0072ad166c7b12323cbdd1d6d9)

I am keeping the conditional use based on the model, otherwise, it would be complaining about unused imports

File: core/embed/firmware/header.S
Comment (grdddj) [reply core/embed/firmware/header.S:29]: So in the end all the changes in `header.S`, `startup.S`, `SConscript.firmware` etc. - not touching emulator - should end up in a separate commit?

Comment (grdddj) [reply core/embed/firmware/header.S]: Done in [91b1938](https://github.com/trezor/trezor-firmware/pull/2230/commits/91b1938c506c2ea02489c5c4ee561eab3cbe1f4b) - during squashing, I would transfer all non-emulator changes into this commit

**Review by matejcik — APPROVED** [review]
LGTM now. please squash & rebase, ideally moving non-emulator changes to their own commit, as you propose

**Comment by grdddj** [comment]
Rebased, squashed, split into relevant commits, included changelogs

File: core/SConscript.unix
Comment (TychoVrahe) [inline core/SConscript.unix:150]: Any reason to use double quotes here unlike rest of the file?

File: core/SConscript.firmware
Comment (TychoVrahe) [inline core/SConscript.firmware:154]: Any reason to use double quotes here unlike rest of the file?

**Review by TychoVrahe — COMMENTED** [review]
Single/double quote usage in SConsrcipt files: inline

**Comment by prusnak** [comment]
> Single/double quote usage in SConsrcipt files: inline

Once this is merged we should run `black core/SConscript.* core/SConstruct` and add that to the Python style-check.

**Comment by matejcik** [comment]
ready to merge after tomorrow's release branching

**Pull request merged**
